### PR TITLE
[terra-search-field] Update search field focus indicator to standard dashed line

### PR DESCRIPTION
**Pull request by trandrew1023**
### Summary

**What was changed:**
Changed the search field's focus indicators to standard Terra dashed line for Terra themes. For Orion Fusion theme, there is still ongoing discussion about the styling so the only change for Orion Fusion is adding the box shadow to the input field to match the buttons to increase the visibility of the focus indicator.

**Why it was changed:**
Focus indicator consistency with other components

### Testing

This change was tested using:

- [x] WDIO
- [ ] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

Terra Default:
image
image
image

Terra Lowlight:
image
image
image

Orion Fusion:
image
image
image

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-8109

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-search-field/src/SearchField.module.scss [hunk 1]
```diff
@@ -47,7 +47,9 @@
       }
 
       &:focus {
-        border-color: var(--terra-search-field-button-focus-border-color, #26a2e5);
+        box-shadow: var(--terra-search-field-clear-focus-box-shadow);
+        outline: var(--terra-search-field-focus-keyboard-outline, 2px dashed #000);
+        outline-offset: var(--terra-search-field-focus-keyboard-outline-offset, 1px);
```
Comment (cm9361) [inline packages/terra-search-field/src/SearchField.module.scss:52]: This pattern is showing up a lot in our code with the accessibility updates.  Should this be converted to a mixin that can be reused or some other reusable concept?

File: packages/terra-search-field/src/SearchField.module.scss [hunk 2]
```diff
@@ -47,7 +47,9 @@
       }
 
       &:focus {
-        border-color: var(--terra-search-field-button-focus-border-color, #26a2e5);
+        box-shadow: var(--terra-search-field-clear-focus-box-shadow);
+        outline: var(--terra-search-field-focus-keyboard-outline, 2px dashed #000);
```
Comment (cm9361) [inline packages/terra-search-field/src/SearchField.module.scss:51]: In this scenario, can't we create one descendant focus pseudo selector to cover all the use cases under the search field?  I don't think we should have to add a selector for each type of element.

File: packages/terra-search-field/tests/wdio/search-field-spec.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ Terra.describeViewports('Search Field', ['medium'], () => {
     });
 
     it('should enter a search term', () => {
-      $('input').setValue('Lore');
+      $('input').setValue('Lore', { mismatchTolerance: 0.1 });
```
Comment (sdadn) [inline packages/terra-search-field/tests/wdio/search-field-spec.js:12]: Out of curiosity, what is the reason for the mismatchTolerance being set??

**Review by sdadn — APPROVED** [review]

File: packages/terra-search-field/tests/wdio/search-field-spec.js (hunk 3 above)
Comment (trandrew1023) [reply packages/terra-search-field/tests/wdio/search-field-spec.js:12]: Those tests were randomly failing due to some of the edges of the focus dotted lines being darker/lighter than the reference

image

**Review by sycombs — APPROVED** [review]

File: packages/terra-search-field/src/SearchField.module.scss (hunk 1 above)
Comment (trandrew1023) [reply packages/terra-search-field/src/SearchField.module.scss:53]: Discussed offline with Charles and team: will need to discuss as team the default focus indicators for each theme and go forward strategy to implement in a way that all components will receive a consistent styling

File: packages/terra-search-field/src/SearchField.module.scss (hunk 2 above)
Comment (trandrew1023) [reply packages/terra-search-field/src/SearchField.module.scss]: @cm9361 I was able to add the styles under the *.focus here https://github.com/cerner/terra-core/pull/3834/commits/4e9be890f0c2087ae529bf141f65a0ce25e667b7

Not sure if there's another approach you would apply here outside of the discussion had about having a library wide focus indicator style

**Review by artpark — APPROVED** [review]

**Review by sycombs — APPROVED** [review]

File: packages/terra-search-field/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Changed
+  * Updated focus indicators to standard dashed line.
```
Comment (sdadn) [inline packages/terra-search-field/CHANGELOG.md:6]:
```suggestion
  * Updated focus indicators to be a standard dashed line.
```

File: packages/terra-search-field/tests/wdio/search-field-spec.js (hunk 3 above)
Comment (sdadn) [reply packages/terra-search-field/tests/wdio/search-field-spec.js:12]: That makes sense. Thanks!

Can you add a comment indicating why this was added for documentation purposes? Something like:
`Setting the mismatchTolerance to 0.1 to account for subtle variances in the dotted line.`

(Or whatever wording is more accurate)

File: packages/terra-search-field/CHANGELOG.md (hunk 4 above)
Comment (trandrew1023) [reply packages/terra-search-field/CHANGELOG.md]: Updated changelog here: http://localhost:8080/tests/cerner-terra-framework-docs/slide-panel-manager/slide-panel-manager-a-11-y-example

File: packages/terra-search-field/tests/wdio/search-field-spec.js (hunk 3 above)
Comment (trandrew1023) [reply packages/terra-search-field/tests/wdio/search-field-spec.js]: Good idea! Added comments for the new mismatchTolerances here https://github.com/cerner/terra-core/pull/3834/commits/0ef9818f16cb8cffc519be4a3e2169e11bfbba28

**Review by eawww — APPROVED** [review]
Beautiful! 😙👌

File: packages/terra-search-field/src/orion-fusion-theme/SearchField.module.scss [hunk 5]
```diff
@@ -52,11 +54,10 @@
...
     --terra-search-field-input-focus-background-color: #fff;
     --terra-search-field-input-focus-border: 1px solid #4cb2e9;
-    --terra-search-field-input-focus-box-shadow: 0 0 8px #a6d9f4;
+    --terra-search-field-input-focus-box-shadow: inset 0 3rem 1.5rem -2rem #f4f4f4, 0 0 1px 1px rgba(255, 255, 255, 0.65), 0 0 2px 3px rgba(76, 178, 233, 0.5), 0 0 6px 4px rgba(76, 178, 233, 0.3);
     --terra-search-field-input-focus-color: #1c1f21;
-
-    --terra-search-field-input-focus-ltr-box-shadow: 0 0 8px #a6d9f4;
-    --terra-search-field-input-focus-rtl-box-shadow: 0 0 8px #a6d9f4;
+    --terra-search-field-input-focus-keyboard-outline: none;
+    --terra-search-input-field-focus-keyboard-outline-offset: none;
```
[the remark this answers is not in the document]
Comment (trandrew1023) [reply packages/terra-search-field/src/orion-fusion-theme/SearchField.module.scss]: Good catch, these are no longer needed and have been removed here https://github.com/cerner/terra-core/pull/3834/commits/d61bf2cf4fa068278c20faed7e9c30ec3821a44e

**Pull request merged**